### PR TITLE
snakemake workflow

### PR DESCRIPTION
**Pull request by nbokulich**
Added a snakemake workflow and brief tutorial.

###  Questions:
1. Should I place example data or outputs in this directory? (I assume not)
2. Are @gregcaporaso 's test data public? or should we point to other test data?

### Notes for the future:
1. maybe give an example using some real data, or work into a full tutorial
2. add @ebolyen 's visualizer to the workflow (is this still in progress?)

### CODE REVIEW

**Comment by thermokarst** [comment]
Do we want the GHA CI to do anything with this snakemake workflow? We could run it as part of the build process (assuming we get example data).

**Comment by gregcaporaso** [comment]
@nbokulich, this looks great! Thanks for taking the time to learn Snakemake and put this together! I plan to download and test in a little bit. Want me to update to use [the new action names](https://github.com/caporaso-lab/genome-sampler/pull/17) when I do that? 

@nbokulich, @thermokarst - answers to your questions:

> Should I place example data or outputs in this directory? (I assume not)

No, let's not put it in the repo right now. 

> Are @gregcaporaso 's test data public? or should we point to other test data?

Most likely yes. I had previously brought up using these data and making them publicly accessible on the ACGU calls and they approved. Now that I've actually assembled the data sets and know exactly what we'd be making public, I'm going to run it by everyone on this afternoon's call and confirm that there are no issues. 

> maybe give an example using some real data, or work into a full tutorial

This is all real data, so I think it should work for a full tutorial. It's more locally-focused than the ACGU pre-print data, but I think it should still be interesting (we don't have permission to use the global dataset from GISAID). @ebolyen noticed an issue in our workflow yesterday which we're going to fix - that should result in a lot more context sequences being retained (just in case you noticed that only two are retained in this data set and that was leading you to think we should use different data). 

> add @ebolyen 's visualizer to the workflow (is this still in progress?)

Yep, still in progress. 

> Do we want the GHA CI to do anything with this snakemake workflow? We could run it as part of the build process (assuming we get example data).

That'd be great. I'll follow up when I have final approval from ACGU to use the data. This can happen post pre-print/MVP. I'm going to submit a PR today to update the main tutorial. Do you guys think I should merge @nbokulich's README.md into that for example as a quickstart section, and then the more detailed tutorial could follow? In that case just running the main tutorial via the GHA CI would cover this as well.

**Comment by thermokarst** [comment]
> @thermokarst - answers to your questions:

Thanks!

> Do you guys think I should merge @nbokulich's README.md into that for example as a quickstart section, and then the more detailed tutorial could follow?

Yes, I think that's a great idea.

File: snakemake/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+## 3. Run the workflow
+
+From within the `/genome-sampler/snakemake/` directory, run the following command:
+```
+snakemake
+```
+
+And watch the magic happen.
+
+The fully processed and subsampled sequences will be output to `sequences.qza`. All other intermediate files will be saved to the same directory.
```
Comment (gregcaporaso) [inline snakemake/README.md:24]: @nbokulich, I think I forgot to mention that we want this to output a fasta file at the end (I forgot to include that in the document that you were reviewing). This way, someone who isn't interested in the QIIME 2 aspects of this doesn't need to know anything about import/export/qza. I'm testing this locally now, and updated the action names in my local copy. I'll add an export step in there too.

Comment (nbokulich) [reply snakemake/README.md:24]: makes sense, thanks!

**Comment by gregcaporaso** [comment]
@nbokulich, I made a few changes here. If you're good with these changes, I think we should merge this and then make some final tweaks when we get the final steps of the workflow together.

**Comment by nbokulich** [comment]
Thanks @gregcaporaso ! One final change to make your edits stick.

> Do you guys think I should merge @nbokulich's README.md into that for example as a quickstart section, and then the more detailed tutorial could follow?

Is this something we should do in this PR? Or follow up later?

Otherwise I am ready to merge when you are.

**Comment by thermokarst** [comment]
I just merged in the latest changes from `master` (the diff was lookin' a little wild).

**Comment by gregcaporaso** [comment]
> quickstart section

@nbokulich, I was thinking of addressing that in the PR when I work on the tutorial document.

**Pull request merged**